docs/constraints/Cplusplus.md

```diff
@@ -0,0 +1,41 @@
+# C++ Constraints
+
+These are the C++ specific constraints applied to the MetaModel before reaching the presets. [Read here to get a more general idea on the overall process](../input-processing.md) of converting a MetaModel to a ConstrainedMetaModel.
+
+## Model Naming
+These are the constraints that is applied to model naming. The `Rule key` is what you can use in the options to overwrite the default behavior. See [constraint customization](./README.md#Customization).
+
+|Rule key|Rule|Resolution|
+|---|---|---|
+|NO_SPECIAL_CHAR|No special characters| Special characters are replaced by their name, for example `!` is replaced with `exclamation`. For C++ `_` and `$` are an exception to this rule. |
+|NO_NUMBER_START_CHAR|No numbers as starting characters|Default behavior is pre pending `number_` in front of the first character|
+|NO_EMPTY_VALUE|No empty values|Default behavior is to use `empty` as name. |
+|NO_RESERVED_KEYWORDS|No reserved keywords|C++ has a list of reserved keywords ([see the full list here](../../src/generators/cplusplus/Constants.ts))|
+|NAMING_FORMATTER|Must be formatted equally|Model name is formatted using pascal case|
```

Comment:
The model namings use `toSnakeCase` function not pascal case, please double check that the implementation and docs (for all the constraints) are matching: https://github.com/asyncapi/modelina/blob/cbf55179f37adaa9b67564b2fc022615d7c66095/src/generators/cplusplus/constrainer/EnumConstrainer.ts#L42C35-L42C46